engine/access/state_stream/state_stream.go

```diff
@@ -45,6 +45,9 @@ type API interface {
 	SubscribeEvents(ctx context.Context, startBlockID flow.Identifier, startHeight uint64, filter EventFilter) Subscription
 	// GetRegisterValues returns register values for a set of register IDs at the provided block height.
 	GetRegisterValues(registerIDs flow.RegisterIDs, height uint64) ([]flow.RegisterValue, error)
+	// SubscribeAccountStatuses subscribes to the streaming of account status changes starting from a specific block ID
+	// and block height, with an optional status filter.
+	SubscribeAccountStatuses(ctx context.Context, startBlockID flow.Identifier, startHeight uint64, filter EventFilter) Subscription
```

Comment:
let's create matching `FromStartBlockID`, etc methods on the backend, then you can simplify the block checking logic. You can reuse some of the logic from the block streaming PR